browser/ui/views/brave_actions/brave_actions_container.cc

```diff
@@ -209,7 +209,7 @@ void BraveActionsContainer::AddAction(const extensions::Extension* extension) {
     // If we do require notifications when popups are open or closed,
     // then we should inherit and pass |this| through.
     actions_[id].view_controller_ = BraveActionViewController::Create(
-        extension->id(), browser_, empty_extensions_container_.get());
+        extension->id(), browser_, empty_extensions_container_.get(), this);
```

Comment:
can we remove a layer of this by passing `brave_action_api_` instead of `this` and making that the delegate?

Reply:
code was removed as we catching popup close using js extension function call.